auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -640,6 +640,24 @@ public void deleteMachineUserAccessKeys(String actorCrn, String accountId, Strin
         }
     }
 
+    /**
+     * Check that machine user has a specific access key in UMS
+     * @param actorCrn actor for the machine user request
+     * @param accountId the account ID
+     * @param machineUserCrn machine user crn that own the access key
+     * @param accessKeyId access key id that we need to check
+     * @return result that is true if the machine user has the queried access key in UMS
+     */
+    public boolean doesMachineUserHasAccessKey(String actorCrn, String accountId,
+            String machineUserCrn, String accessKeyId) {
+        try (ManagedChannelWrapper channelWrapper = makeWrapper()) {
+            UmsClient client = makeClient(channelWrapper.getChannel(), actorCrn);
+            String requestId = UUID.randomUUID().toString();
+            List<String> accessKeys = client.listMachineUserAccessKeys(requestId, actorCrn, accountId, machineUserCrn, true);
```

Comment:
Isn't there a cheaper method of validating wether and access key exists works?
GetAccessKeyVerificationData? I'm not sure it's better, but listing all keys seems tedious.

Reply:
Also what happens if someone deletes the key or deactivates it? I'm not sure this check by list solves that problem.

Reply:
that's only listing the access keys that are assigned to the machine user + we need to first check the machine user existence itself anyway + with this no any new calls added. so i think that should be fine for now (can be refactored later)